libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/comboChart/PluggableComboChart.tsx

```diff
@@ -235,6 +244,76 @@ export class PluggableComboChart extends PluggableBaseChart {
         );
     }
 
+    protected getDefaultAndAvailableSort(
+        buckets: IBucketOfFun[],
+        properties: IVisualizationProperties,
+    ): {
+        defaultSort: ISortConfig["currentSort"];
+        availableSorts: ISortConfig["availableSorts"];
+    } {
+        const measures = getBucketItemsByType(buckets, BucketNames.MEASURES, [METRIC]);
+        const secondaryMeasures = getBucketItemsByType(buckets, BucketNames.SECONDARY_MEASURES, [METRIC]);
+        const viewBy = getBucketItems(buckets, BucketNames.VIEW);
+        const stackMeasures = properties?.controls?.stackMeasures;
+        const defaultSort = viewBy.map((vb) => newAttributeSort(vb.localIdentifier, "asc"));
+
+        if (!isEmpty(viewBy) && (!isEmpty(measures) || !isEmpty(secondaryMeasures))) {
+            const mergedMeasures = [...measures, ...secondaryMeasures];
+            const areaSortEnabled = mergedMeasures.length > 1 ? true : false;
+
+            if (stackMeasures) {
+                return {
+                    defaultSort,
+                    availableSorts: [
+                        {
+                            itemId: localIdRef(viewBy[0].localIdentifier),
+                            attributeSort: {
+                                areaSortEnabled: true,
+                                normalSortEnabled: true,
+                            },
+                        },
+                    ],
+                };
+            }
+
+            return {
+                defaultSort,
+                availableSorts: [
+                    {
+                        itemId: localIdRef(viewBy[0].localIdentifier),
+                        attributeSort: {
+                            areaSortEnabled,
+                            normalSortEnabled: true,
+                        },
+                        metricSorts: [
```

Comment:
Why the array copying? Isn't `metricSorts: mergedMeasures.map((m) => newMeasureSortSuggestion(m.localIdentifier))` enough? The `map` function already creates a new array.